src/routes/user.router.js

```
@@ -0,0 +1,31 @@
+import express from 'express';
+import { userController } from '../controllers/user.controller.js';
+import { catchError } from '../utils/catchError.js';
+import { isAuth } from '../midlewares/isAuth.js';
+import { isNotAuth } from '../midlewares/isNotAuth.js';
+
+export const userRouter = express.Router();
+
+userRouter.post(
+  '/',
+  catchError(isNotAuth),
+  catchError(userController.createUser),
+);
+
+userRouter.get(
+  '/:userId',
+  catchError(isAuth),
+  catchError(userController.getUserInfo),
+);
+
+userRouter.patch(
+  '/:userId',
+  catchError(isAuth),
+  catchError(userController.changeUserName),
+);
+
+userRouter.delete(
+  '/:userId',
+  catchError(isAuth),
+  catchError(userController.deleteUser),
+);
```

Consider validating the `:userId` param as a number earlier (router-level or middleware) to fail fast on malformed requests. Although controllers perform some validation, a simple numeric check in middleware improves error clarity and avoids unnecessary DB lookups.